manta-crypto/src/eclair/bool.rs

```diff
@@ -96,6 +98,47 @@ where
 {
     /// Selects `true_value` when `bit == true` and `false_value` when `bit == false`.
     fn select(bit: &Bool<COM>, true_value: &Self, false_value: &Self, compiler: &mut COM) -> Self;
+
+    /// Selects an element from `table` by repeated iteration of `select` over `bits`.
+    /// The `bits` are ordered from most significant to least significant, forming unsigned integers in binary representation
+    /// which are understood as the `table` indices.
+    #[inline]
+    fn select_from_table<'s, B, T>(bits: B, table: T, compiler: &mut COM) -> Self
```

Comment:
Check if https://docs.rs/ark-r1cs-std/latest/ark_r1cs_std/select/trait.CondSelectGadget.html#method.conditionally_select_power_of_two_vector is faster/slower than our implementation.

Reply:
Mathematical count is the same. Also, the primitive test

/// Runs our impl of table selection against arkwork's.
    #[test]
    fn constraints_per_algorithm() {
        let mut cs = Compiler::<Bls12_381_Edwards>::for_proofs();
        let scalar = Scalar::<Bls12_381_Edwards>::gen(&mut OsRng);
        let base = Group::<Bls12_381_Edwards>::sample((), &mut OsRng);
        let base_var =
            base.as_known::<Secret, GroupVar<Bls12_381_Edwards, AffineVar<_, _>>>(&mut cs);
        let scalar_var =
            scalar.as_known::<Secret, ScalarVar<Bls12_381_Edwards, AffineVar<_, _>>>(&mut cs);
        let window_size = 10usize;
        let window = Window::new(window_size, base_var, & mut cs).into_inner();
        let chunk = ToBitsGadget::to_bits_be(&scalar_var.0).unwrap();
        let bits = chunk.chunks_exact(window_size).next().unwrap();
        let window_2 = window.iter().map(|x| x.clone()).map(|x| x.0).collect::<Vec<_>>();
        let ctr1 = cs.constraint_count();
        let _my_algo = ConditionalSelect::select_from_table(bits, &window, & mut cs);
        let ctr2 = cs.constraint_count();
        let _their_algo = CondSelectGadget::conditionally_select_power_of_two_vector(bits, &window_2);
        let ctr3 = cs.constraint_count();
        assert!(cs.is_satisfied());
        println!("My algo: {:?}", ctr2 - ctr1);
        println!("Their algo: {:?}", ctr3 - ctr2);
    }
    
    returns the same number of constraints for both table look-ups. It seems it amounts to the same thing.